File: buck_tanley_app/lib/models/dto/UserDTO.dart
import 'package:buck_tanley_app/models/entity/User.dart';

class UserDTO {
  final String userId;
  final String nickname;
  final String? image;
  final String introduction;
  final bool gender;
  final int age;
  final int status;

  UserDTO({
    required this.userId,
    required this.nickname,
    required this.image,
    required this.introduction,
    required this.gender,
    required this.age,
    required this.status,
  });

  factory UserDTO.init(String userId) {
    return UserDTO(
      userId: userId,
      nickname: "",
      image: null,
      introduction: "",
      gender: false,
      age: 20,
      status: 2,
    );
  }

  factory UserDTO.fromUser(User? user) {
    if (user == null) {
      return UserDTO.init("");
    }
    return UserDTO(
      userId: user.userId,
      nickname: user.nickname,
      image: user.image,
      introduction: user.introduction,
      gender: user.gender,
      age: user.age,
      status: user.status,
    );
  }

  // JSON -> 객체 변환
  factory UserDTO.fromJson(Map<String, dynamic> json) {
    return UserDTO(
      userId: json['userId'],
      nickname: json['nickname'],
      image: json['image'],
      introduction: json['introduction'],
      gender: json['gender'],
      age: json['age'],
      status: json['status'],
    );
  }

  // 객체 -> JSON 변환
  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'nickname': nickname,
      'image': image,
      'introduction': introduction,
      'gender': gender,
      'age': age,
      'status': status,
    };
  }
}

File: buck_tanley/src/main/java/com/example/buck_tanley/domain/dto/UserDTO.java
package com.example.buck_tanley.domain.dto;

import java.util.Objects;

import com.example.buck_tanley.domain.entity.User;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UserDTO {
  private String userId;
  private String nickname;
  private String image;
  private String introduction;
  private Boolean gender;
  private Short age;
  private Short status; // 0 : 현재 비활성 상태, 1 : 활성 상태, 2 : 매칭 중인 상태, 3 : 매칭 된 상태

  public UserDTO(User user) {
    this.userId = user.getUserId();
    this.nickname = user.getNickname();
    this.image = user.getImage();
    this.introduction = user.getIntroduction();
    this.gender = user.getGender();
    this.age = user.getAge();
    this.status = user.getStatus();
  }

  public UserDTO(String userId) {
    this.userId = userId;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o)
      return true;
    if (o == null || getClass() != o.getClass())
      return false;
    UserDTO userDTO = (UserDTO) o;
    return Objects.equals(userId, userDTO.userId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(userId);
  }
}